hazelcast/src/main/java/com/hazelcast/internal/metrics/MetricsUtil.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.metrics;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import javax.annotation.Nonnull;
+import java.util.AbstractMap.SimpleImmutableEntry;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * A utility to escape and parse metric key in tagged format:<pre>
+ *     [tag1=value1,tag2=value2,...]
+ * </pre>
+ */
+public final class MetricsUtil {
```

Comment:
Does util need to be in the main metrics package where the API is defined, or is this an implementation detail and can it be moved to metrics.impl ?

Reply:
It will be used by MC. I consider it a public API of metrics. It will escape values in a particular way and provides parse method that is only used externally.

Reply:
AFAIK MC doesn't share any HZ code.

Reply:
The new MC will, it will use Jet client to connect, even to get the metrics

Reply:
Ok. But in the not so far future, JMC == MC. And since MC has not access to HZ sources, JMC will not have access to HZ source. 

Or is my understanding not correct?